src/csharp/Grpc.Core/Internal/AsyncCall.cs

```diff
@@ -498,8 +507,12 @@ private void HandleFinished(bool success, ClientSideStatus receivedStatus)
                 ReleaseResourcesIfPossible();
             }
 
-            var status = receivedStatus.Status;
+            if (delayedStreamingWriteTcs != null)
```

Comment:
I think it would work fine without it, but changed to store the tcs into a local variable under a lock to make sure it's 100% clean.